pmd-core/src/main/java/net/sourceforge/pmd/renderers/HTMLRenderer.java

```diff
@@ -48,6 +48,19 @@ public class HTMLRenderer extends AbstractIncrementingRenderer {
     private int violationCount = 1;
     boolean colorize = true;
 
+    final String style = "<link rel='stylesheet' type='text/css' href='https://cdn.datatables.net/1.12.1/css/jquery.dataTables.css'>" +
+        "<link rel='stylesheet' type='text/css' href='https://cdn.jsdelivr.net/npm/bootstrap@5.2.0/dist/css/bootstrap.min.css'>"
+        ;
+
+        final String scripts = "<script src='https://code.jquery.com/jquery-3.6.1.min.js'></script> " +
+        "<script type='text/javascript' charset='utf8' src='https://cdn.datatables.net/1.12.1/js/jquery.dataTables.js'></script>"
+        ;
```

Comment:
Adding this would require to have internet access when viewing PMD's html report.
I think, it would still be nice to be able to view the report standalone.

Maybe we can leverage https://www.webjars.org/ to add jquery+DataTables as dependencies. The renderer would then output additionally the js-files and css-files and one would need to configure a output directory (similar to what we do with the yet another html renderer -> https://pmd.github.io/latest/pmd_userdocs_report_formats.html#yahtml). Alternatively we could embed the js+css code directly in the html file the renderer produces.